_posts/2018-12-16-advent-darkhttpd.markdown

```diff
@@ -0,0 +1,37 @@
+---
+layout: post
+title: "The Advent of Void: Day 16: darkhttpd"
+comments: true
+---
+
+[darkhttpd](https://unix4lyfe.org/darkhttpd/) is a small webserver.
+Due to it's design choices it is probably not a good fit for serving
+large sites but it is well suited for simple one-off tasks.
+
+To start `darkhttpd`, run `darkhttpd /path/to/wwwroot`.
+
+An exemplary use case, is to distribute xbps packages built on some box
+to more machines for testing, before pushing them to the `void-packages`:
+
+```
+$ darkhttpd hostdir/binpkgs/firefox
+darkhttpd/1.12, copyright (c) 2003-2016 Emil Mikulic.
+listening on: http://0.0.0.0:8080/
+```
+
+`darkhttpd` by defaults listens on port 8080, so to update the testing machines,
+running `sudo xbps-install --repository=http://buildbox:8080 -Suy` is all to be done.
```

Comment:
Have you tried this? I suspect not :P

Reply:
Most of us don't automatically sign our packages, which is why I express doubt.

Reply:
I used it with signed packages. See log below :)

Reply:
added a note that packages must be signed for this to work.